lib/galaxy/tools/parameters/validation.py

```diff
@@ -248,6 +249,23 @@ def validate( self, value, trans=None ):
                 raise ValueError( self.message )
 
 
+class DatasetFilesPathEmptyValidator( Validator ):
+    """Validator that checks if a dataset's files_path exists and is not empty."""
+    def __init__( self, message=None ):
+        self.message = message
+
+    @classmethod
+    def from_element( cls, param, elem ):
+        return cls( elem.get( 'message', None ) )
+
+    def validate( self, value, trans=None ):
+        if value:
+            if not os.path.isdir(value.extra_files_path) or len(os.listdir(value.extra_files_path)) == 0:
```

Comment:
There are some abstractions in galaxy.model.Dataset, could you use those? In particular you have get_size and get_total_size, which in the case of empty extra files would be the same, I think. That also takes care of object_store interaction and caching the result in the database.